Practice-6/NguyenThiHuong/README.md

```diff
@@ -0,0 +1,458 @@
+# DOCKER NETWORK 
+
+## Content 
+
+  - [I. What are docker network modes?]()
+  - [II. How does docker-compose provide networking for its services ]()
+  - [III. How does docker swarm do multi-host networking]()
+  - [References]()
+
+---
+
+## I. What are docker network modes?
+
+In order to connect containers, we have to configure the docker network. There are four docker network modes: 
+
+1. Bridge mode
+2. Host mode
+3. None mode
+4. Container mode
+
+### 1. Bridge mode
+
+Below is an example of bridge network mode.
+
+```
+huong@py:~$ docker network ls
+NETWORK ID     NAME                                                DRIVER    SCOPE
+07d4bc0d9391   bridge                                              bridge    local
+9de1de6483bc   cf-example-manage-mongodb_default                   bridge    local
+3d4c34f26618   composetest_default                                 bridge    local
+65373d504780   docker-compose-prometheus-and-grafana_monitor-net   bridge    local
+
+```
+ 
+- Bridge mode is the default network mode of docker. If you do not write the – net parameter, it is the bridge mode. When docker run – P is used, docker actually makes DNAT rules in iptables to realize port forwarding function..
+- When the docker process starts, by default, a virtual bridge named docker0 will be created on the host, and the docker container started on this host will be connected to the virtual bridge. The virtual bridge works like a physical switch, so that all containers on the host are connected to a layer-2 network through the switch.
+- The Docker bridge driver automatically installs rules in the host machine so that containers on different bridge networks cannot communicate directly with each other.
+- The docker then assigns an IP address from the docker 0 subnet to the container, and set the docker 0 IP address as the default gateway of the container. Create a pair of virtual network card Veth pair devices on the host. Docker puts one end of the Veth pair device in the newly created container and names it eth0 (network card of the container). The other end is placed in the host and named after vethxxx. The network device is added to the docker 0 bridge. It can be viewed through the brctl show command.
```

Comment:
docker 0 -> `docker0`

what is the relationship between `docker0` and the host interface (`enps0`)?